tools/analysis/programs.go

```diff
@@ -145,7 +172,17 @@ func (programs Programs) check(
 						return nil, err
 					}
 
-					elaboration = programs[importedLocation].Checker.Elaboration
+					// If the imported program has a checker, use its elaboration for the import
+					if programs[importedLocation].Checker != nil {
+						elaboration = programs[importedLocation].Checker.Elaboration
+					}
+
+					// If the imported program had an error while loading, record it
+					loadError = programs[importedLocation].LoadError
```

Comment:
Refactor the common subexpressions into single variables:

```suggestion
					program := programs[importedLocation]
					checker := program.Checker

					if checker != nil {
						elaboration = checker.Elaboration
					}

					// If the imported program had an error while loading, record it
					loadError = program.LoadError
```